hack/validate-migration.sh

```diff
@@ -371,6 +371,31 @@ check_apply_in_real_cluster() {
     rm "$apply_logfile"
 }
 
+# Check that the OCI-TA variant of this task also has a migration script
+# Arguments: the normal-task migration file path
+check_oci_ta_migration() {
+    local migration_file=$1
+    # e.g. migration_file=task/foo/0.2/migrations/0.2.1.sh
+    local parent_dir
+    parent_dir=$(resolve_migrations_parent_dir "$migration_file")    # => task/foo/0.2
+
+    # get the “foo” and “0.2”
+    local task_dir="${parent_dir%/*}"                               # => task/foo
+    local task_name
+    task_name=$(basename "$task_dir")                               # => foo
+    local version
+    version=$(basename "$parent_dir")                               # => 0.2
+
+    # path to the OCI-TA variant migrations:
+    local oci_mig_dir="${task_dir}-oci-ta/${version}/migrations"
+
+    if [ ! -d "$oci_mig_dir" ] || [ -z "$(ls -A "$oci_mig_dir")" ]; then
```

Comment:
This checks that `{task}-oci-ta` has at least 1 migration script, which is a good start. But IMO we should ensure that both tasks have the same set of migration scripts.

Using the example from the comments: `migration_file=task/foo/0.2/migrations/0.2.1.sh`
IMO the check should validate that this file also exists: `task/foo-oci-ta/0.2/migrations/0.2.1.sh`

Reply:
You're right; resolved.